### PR TITLE
Add new job

### PR DESCRIPTION
**Pull request by AdamC12**
Link to ticket: https://trello.com/c/dxpoJ3cq/1-1-users-can-add-jobs
After this PR: A homepage displaying jobs is available, with a new job button, which takes you to a form to create a new job
image
image

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
+  resources :jobs
```
Comment (Robfaldo) [inline config/routes.rb:3]: Question: what does this line of code do?

And then a follow up, is it doing more than we need it to (creating unnecessary things) and if so how could we restrict it?

Comment (AdamC12) [reply config/routes.rb:3]: I believe what resources is doing is setting up 7 default rails routes so you don't have to type them all out 

(e.g like get "/jobs/new", to: "jobs#new")

it's definitely creating unecessary rroutes we don't need right now, like patches etc. We could restrict it by using only

e.g resources: jobs, only[:index,:new] and then increase as we go

File: config/routes.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
+  resources :jobs, only: [:index, :new, :create]
```
Comment (Robfaldo) [inline config/routes.rb:3]: awesome 👍

File: app/controllers/jobs_controller.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+end
+
+private_methods
+def job_params
+  params.require(:job).permit(:title,:company,:location,:link)
+end
```
[the remark this answers is not in the document]
Comment (AdamC12) [reply app/controllers/jobs_controller.rb:24]: @Robfaldo what do you mean by 'outside the controller'

File: config/routes.rb (hunk 1 above)
Comment (Robfaldo) [reply config/routes.rb]: Awesome 👍

File: app/controllers/jobs_controller.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (Robfaldo) [reply app/controllers/jobs_controller.rb:24]:
```
class Controller
  # inside the controller
end

# outside the controller 
```

[the remark this answers is not in the document]
Comment (AdamC12) [reply app/controllers/jobs_controller.rb]: pushed a change to make it more readable / better

**Review by Robfaldo — APPROVED** [review]
👍 Nice job. Have tested locally and all works. 

Would prefer if you added a README with instructions for getting started but it's optional

**Review by Robfaldo — APPROVED** [review]

**Pull request merged**